### ISSUE TITLE
PR CI is not building rust-analyzer

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#115031]
In https://github.com/rust-lang/rust/pull/114993 I ran down the totally wrong path for a while when I saw an RA build failure in bors -- PR CI was fine, so I was sure this must have been something odd about the bors builder. Turns out however we just don't check if RA builds, not even if `src/tools/rust-analyzer` is changed by the PR.

I thought the mingw-check builder does at least a check-build of *everything*, but that does not seem to be the case?
@rust-lang/rust-analyzer @rust-lang/infra

**Comment by lnicola** [comment]
I think this is intentional and actually saves quite a bit of CI time. Would it be better if we asked PR authors to not make unneeded changes (like compile fixes) to RA in this repo?

**Comment by RalfJung** [comment]
I think it runs `./x.py check`

https://github.com/rust-lang/rust/blob/4f4dae055b470c889968b02964a28477f9daff62/src/ci/docker/host-x86_64/mingw-check/Dockerfile#L43

but that doesn't include RA. It includes clippy and Miri. RA should probably be added?

**Comment by RalfJung** [comment]
A check-build of RA takes around 15s on my system. I don't think it will be prohibitively slow on CI.

**Comment by RalfJung** [comment]
For other tools like clippy and Miri, things work as follows:
- We always run a check-build on the mingw-check builder. This takes less than a minute per tool so it's fine.
- If they change, the tools builder gets enabled, and it runs their full test suite. This takes a while which is why it doesn't happen for all PRs.

**Comment by Veykril** [comment]
I agree, we should have r-a be built on rust-lang/rust CI. The main point was not running r-a's (complete) test suite I believe when people investigated CI times, which is fine.

**Comment by ozkanonur** [comment]
> But that's not the point here, I think this is asking for a bootstrap change: ./x.py check should also build RA. Cc https://github.com/orgs/rust-lang/teams/bootstrap

We can add all the enabled tools from `tools` in the `config.toml` file to `x check`.

**Comment by onur-ozkan** [comment]
With #115111, if enabled in the configuration, rust-analyzer will be involved in the compilation.
